ScriptableNextBirthdays.js

```diff
@@ -361,25 +372,27 @@ const inputText = Array.isArray(escapedText) ? `['${
 const textLength = new TextLength();
 const lightDotsLength = await textLength.computeLength(settings.textOverflowChar, settings.textSize, settings.textFontFamilies.thin)
 
-async function computeWidget(fm, settings){
-  const widget = new ListWidget();
+async function createWidget(fm, settings){
   const recomputeBirthdays = !config.runsInWidget;
   
-  const allBirthdays = recomputeBirthdays ? await updateAndGetCache(fm) : loadCache(fm);
+  let allBirthdays;
+  try {
+    allBirthdays = recomputeBirthdays ? await updateAndGetCache(fm) : loadCache(fm);
+  } catch {
+    return createErrorWidget(new ContactsAccessDeniedError(), settings);
+  }
   
   let nextContacts;
   nextContacts = computeNextBirthdays(allBirthdays, settings);
-  
   nextContacts = nextContactsAndRelativeDates(nextContacts);
```

Comment:
This is an impressive solution! Good job! 😎 
Did you already have a look at this library? It's already integrated into our project and would take care of this problem for you. 😉